Version 4

rowanmanning opened the pull request:
Each of the commits in here have been reviewed separately, and a [migration guide is available here](https://github.com/Financial-Times/o-labels/tree/4.x#migration-guide).

## Master

<img width="282" alt="screenshot 2019-02-07 at 13 13 56" src="https://user-images.githubusercontent.com/138944/52413820-85f25880-2ada-11e9-85a9-b95cc9a5d4a7.png">

## Internal

<img width="389" alt="screenshot 2019-02-07 at 13 14 37" src="https://user-images.githubusercontent.com/138944/52413821-85f25880-2ada-11e9-8bf5-390ca1ef998b.png">

rowanmanning committed:
Large scale refactor for v4 (#26)

* Large scale refactor for v4

This refactors all of the code for v4, and introduces branding. I've
also added a few new states and removed some old ones but these will be
reviewed properly separately.

Work that this doesn't include, but will be done separately:

  - An overhaul of the README (waiting to know more about which variants
    we will continue to support)
  - Comprehensive SassDoc (I wanted to get an early review in before I
    document everything)
  - Tests (I'd like to try out some testing with True later and we'll
    need a Pa11y test for accessibility)

* Remove unnecessary sassdoc for now

* Rename default brand configurations

* Use scales over pixel values in brand config

* Reduce/simplify demo dependencies

* Remove the need for a private mixin

* Switch padding configurations back to pixels

* Rename back to o-labels

* Move wide styles into beta lifecycle state

rowanmanning committed:
Remove a rogue alpha reference (#27)

rowanmanning committed:
Update the README and add a migration guide (#28)

rowanmanning committed:
Add Sassdoc to all the mixins (#29)

rowanmanning committed:
Add true tests for padding calculation (#31)

rowanmanning committed:
Design and variant overhaul (#32)

* Design and variant overhaul

This removes the small variant, as the standard size is now smaller. The
big variant is now around the same size as the previous standard size.

All labels are now transformed to uppercase.

* Updates based on review feedback

rowanmanning committed:
Oh look, the small variant is back! (#33)

* Oh look, the small variant is back!

I wonder whether we'll get to delete and re-add it again! 😄

* Simplify demo language

rowanmanning committed:
Add a Pa11y demo (#34)

This currently doesn't test everything - I've done that manually. But
once OBT supports a11y testing for all brands this should just start
working.

rowanmanning committed:
Move to a more oTable way of doing mixins (#35)

We now have a single oLabels mixin which outputs everything, similar to
the main oTable mixin.

JakeChampion commented:
Is this waiting on approval from someone other than @Financial-Times/origami-core ?

apaleslimghost commented:
the App is currently using the `active` and `error` states, although this isn't consistent with ft.com's current labelling of live blogs, so this is probably a good point for it to not be using those labels. at one point the `live` (and possibly `pending`) state was being used _[somewhere on ft.com trying to find out]_ but i don't think they are any more.

rowanmanning committed:
Make the small variant a little smaller (#38)

This is to address issues with the small label when lots are displayed
on the same page, e.g. in MyFT. The new padding more closely matches the
original label padding and so fits nicely with the existing designs.

rowanmanning committed:
Remove default capitalisation (#39)

The only label that's now capitalised is the beta label, the rest are
lowercase at the moment. We may add capitalisation as a variant later if
there are compelling use-cases for it.

rowanmanning commented:
This is still needing some small changes.
- [x] The screenshot needs updating
- [x] The main `oLabels` mixin needs to have a single map option
- [x] Make changes to the state mixin based on discussion with the team
- [x] Move to a separate migration guide

gvonkoss commented on src/scss/_mixins.scss:
```diff
@@ -0,0 +1,108 @@
+/// @example scss - Base styles, big size variant, and gold tier state variant
+///   @include oLabels(('big'), ('tier-gold'));
+/// @access public
+@mixin oLabels($sizes: $_o-labels-sizes, $states: $_o-labels-states) {
```
@notlee and I have chatted about file structure and such before  — we've agreed to put the primary mixin in the `main.scss` file, in order to make it easier to find and consistent across our components 😬

notlee commented:
New labels with controls on Heimdall https://heimdall.in.ft.com/dashboard?teamid=origami-team

<img width="825" alt="screenshot 2018-12-14 at 13 53 57" src="https://user-images.githubusercontent.com/10405691/50007110-b8367c80-ffa7-11e8-8cbe-4e22e3d6375b.png">

notlee commented:
More labels! Lodestar.

<img width="190" alt="screenshot 2019-01-25 at 11 37 29" src="https://user-images.githubusercontent.com/10405691/51743778-a6abbe80-2095-11e9-910e-88d1ec1501d2.png">

rowanmanning committed:
Switch the primary mixin to have a single argument

This also moves the primary mixin into the main.scss file, to match the
components which have had more recent releases.

rowanmanning committed:
Move to an external migration guide

The README and migration guide now conform to the standard set by other
recent components.

rowanmanning committed:
Rename oLabelsState to oLabelsAddState

Based on a discussion between Gabi, Lee, and I. We've decided we should
be explicit that you're _adding_ a new thing rather than customising an
existing thing.
